ui/pages/swaps/list-with-search/list-with-search.js

```diff
@@ -0,0 +1,180 @@
+import { isValidHexAddress } from '../../../../shared/modules/hexstring-utils';
+import { I18nContext } from '../../../contexts/i18n';
+import { fetchToken } from '../swaps.util';
+import { getCurrentChainId } from '../../../selectors/selectors';
+
+let timeoutIdForSearch;
+
+export default function ListWithSearch({
+  itemsToSearch = [],
+  listTitle,
+  maxListItems,
+  onClickItem,
+  onOpenImportTokenModalClick,
+  shouldSearchForImports,
+  Placeholder,
+  hideRightLabels,
+  hideItemIf,
+  listContainerClassName,
+  searchQuery,
+  setSearchQuery,
+}) {
+  const itemListRef = useRef();
+  const t = useContext(I18nContext);
+
+  const [items, setItems] = useState(itemsToSearch);
+  const chainId = useSelector(getCurrentChainId);
+
+  /**
+   * Search a custom token for import based on a contract address.
+   *
+   * @param {string} contractAddress
+   */
+  const handleSearchTokenForImport = async (contractAddress) => {
+    try {
+      const token = await fetchToken(contractAddress, chainId);
+      if (token) {
+        token.primaryLabel = token.symbol;
+        token.secondaryLabel = token.name;
+        token.notImported = true;
+        setItems([token]);
+        return;
+      }
+    } catch (e) {
+      log.error('Token not found, show 0 results.', e);
+    }
+    setItems([]); // No token for import found.
+  };
+
+  const handleSearch = async (newSearchQuery) => {
+    setSearchQuery(newSearchQuery);
+    if (timeoutIdForSearch) {
+      clearTimeout(timeoutIdForSearch);
+    }
+    timeoutIdForSearch = setTimeout(async () => {
+      timeoutIdForSearch = null;
+      const trimmedNewSearchQuery = newSearchQuery.trim();
+      const trimmedNewSearchQueryUpperCase =
+        trimmedNewSearchQuery.toUpperCase();
+      const trimmedNewSearchQueryLowerCase =
+        trimmedNewSearchQuery.toLowerCase();
+      if (!trimmedNewSearchQuery) {
+        setItems(itemsToSearch);
+        return;
+      }
+      const validHexAddress = isValidHexAddress(trimmedNewSearchQuery);
+      let filteredItems = [];
+      if (validHexAddress) {
+        // E.g. DAI token: 0x6b175474e89094c44da98b954eedeac495271D0f
+        const foundItem = itemsToSearch.find((item) => {
+          return item.address === trimmedNewSearchQueryLowerCase;
+        });
+        if (foundItem) {
+          filteredItems.push(foundItem);
+        }
+      } else {
+        filteredItems = filter(itemsToSearch, function (item) {
+          return item.symbol.includes(trimmedNewSearchQueryUpperCase);
+        });
+      }
+      const results = newSearchQuery === '' ? itemsToSearch : filteredItems;
+      if (shouldSearchForImports && results.length === 0 && validHexAddress) {
+        await handleSearchTokenForImport(trimmedNewSearchQuery);
+        return;
+      }
+      setItems(results);
+    }, 350);
+  };
+
+  useEffect(() => {
+    handleSearch(searchQuery);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is this eslint line necessary?

Reply:
It only silences a warning, but code works without it as well, so I've removed it.

When I pass the `handleSearch` dependency to remove the warning, it triggers this effect many times. That's why I didn't add it into the dependency list. Maybe there is a better way where the `handleSearch` fn doesn't trigger this effect.